modules/google-maps/src/utils.js

```diff
@@ -0,0 +1,119 @@
+/* global document, google */
+import {Deck} from '@deck.gl/core';
+
+export function createDeckInstance(map) {
+  const container = map.getDiv();
+
+  const deckCanvas = document.createElement('canvas');
+  container.appendChild(deckCanvas);
+  Object.assign(deckCanvas.style, {
+    // map container position is always non-static
+    position: 'absolute',
+    pointerEvents: 'none',
+    left: 0,
+    top: 0,
+    width: '100%',
+    height: '100%'
+  });
```

Comment:
I've taken a different approach in [this gist](https://gist.github.com/donmccurdy/61f94f91b60682bf325d2cceae4c70f2#file-googlemapslayer-js-L70-L75):

```js
const overlayLayerEl = this.overlay.getPanes().overlayLayer;
overlayLayerEl.appendChild(this.canvasEl);
```

The various `overlay.getPanes()` divs are the attachment points that Maps API encourages developers to use, and should fix the zIndex and fullscreen mode issues. Unfortunately none of the panes have a fixed position relative to the page, so the canvas's CSS position must be changed on every frame as shown in that gist. I don't love that solution, but I haven't found any problems with it yet either.